src/transforms/mapperTransforms.ts

```diff
@@ -169,18 +185,88 @@ function transformPrimitiveMapper(pipelineValue: PipelineValue) {
   };
 }
 
+// TODO: Make sure this is the correct way to handle additionalProperties
+function getAdditionalProperties(
+  immediateParents: Schema[]
+): Mapper | undefined {
+  return immediateParents.some(p => p.type === SchemaType.Dictionary)
+    ? {
+        type: {
+          name: MapperType.Object
+        }
+      }
+    : undefined;
+}
+
+function checkUberParent(objectSchema: ObjectSchema) {
```

Comment:
Is it possible you could change this to `isUberParent` and then move the caching step to `transformObjectMapper` which seems to be the only place this gets used?  Or maybe I'm missing a use case?